### DISCUSSION TITLE
window-post-miner的集群化设计如何实现

### DISCUSSION THREAD

**Discussion by Zengdewei1** [filecoin-project/community-china#77] [存储方案]
一台机器上扇区非常多，window-post计算不过来，能否让其他机器帮忙进行window-post的计算，比如构建一个分布式集群

**Comment by summerxue1992** [comment]
从浏览器上看，同一个deadline可以多次提交window post，且每天有48个deadline窗口，所以你是有多少扇区，面临这个问题。 这几天也思考了这个问题，现在我们大概6p左右算力的集群，封装满的时候，认为暂时不用考虑此问题

**Reply by Zengdewei1** [reply]
> 且
一个deadline里面，一个分区有2000个扇区需要封装，约0.06P，一个扇区window-post需要2s，那么证明需要4000s，约67分钟。一个deadline是30分钟。

**Reply by Zengdewei1** [reply]
> 从浏览器上看，同一个deadline可以多次提交window post，且每天有48个deadline窗口，所以你是有多少扇区，面临这个问题。 这几天也思考了这个问题，现在我们大概6p左右算力的集群，封装满的时候，认为暂时不用考虑此问题

你能否大致计算下你们6P，是怎么在规定时间内完成封装的。

**Reply by summerxue1992** [reply]
我们6P集群刚开始封装。现在封装了几百t。  但是一个deadline已经有了2300个扇区了，不是你说的 2000个。还有就是一个扇区window-post计算需要2s这个是求证过的吗。这样的话，2000个扇区的deadline 提交时空证明，怎么能成功呢

**Reply by summerxue1992** [reply]
我们是64G扇区

**Reply by Zengdewei1** [reply]
你说的对，我应该求证下，window-post具体需要多少时间。一个deadline一个partition，最大2349个扇区，64G扇区就是147T。不同的deadline是对应不同的分区吗？你们现在在一个deadline里面可以成功提交证明吗？你们的机器配置如何呢？特别是GPU。

**Reply by summerxue1992** [reply]
我们2300扇区的那个deadline，今天提交成功了的。 gpu  2080Ti

**Reply by Zengdewei1** [reply]
那么一个deadline147T，48个deadline就是6.89P。你们的集群完全可以负载。不用多台window-post的机器。可能我提的这个确实是个伪需求

**Reply by summerxue1992** [reply]
我看到浏览器上别人大的矿工同一时间可以提交多次时空证明，所以这个还得再看看，是不是不存在这个瓶颈

**Reply by Zengdewei1** [reply]
是的。一个窗口至少有一个partition，这个是真实存在。那么如果分区足够多的话。比如有超过12P的数据，那就会有96个分区。那么每一轮deadline都会要抽查2个partition。分区更多，每一轮要抽取的也更多，所以window-post可能存在瓶颈

**Comment by zhaobo443** [comment]
小型机房建设，专业IPFS节点搭建，分币系统app定制全套filecoin,包括测ceph存储技术，长期合作全部免费奉送，长期维护只抽取5%服务费！欢迎电话咨询：19965407481
